nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/handler/impl/DefaultBatchHandlerImpl.java

```diff
@@ -0,0 +1,124 @@
+import org.nuxeo.ecm.automation.server.jaxrs.batch.Batch;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchFileEntry;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.AbstractBatchHandler;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.BatchFileInfo;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.transientstore.api.TransientStore;
+import org.nuxeo.ecm.core.transientstore.api.TransientStoreService;
+import org.nuxeo.runtime.api.Framework;
+
+public class DefaultBatchHandlerImpl extends AbstractBatchHandler {
+
+    protected static final Log log = LogFactory.getLog(DefaultBatchHandlerImpl.class);
+
+    protected static final String BATCH_HANDLER_NAME = "default";
+
+    protected static final String PROVIDER_KEY = "provider";
+
+    private static class Config {
+        public static final String TRANSIENT_STORE_NAME = "transientStore";
+    }
+
+    private String transientStoreName;
+
+    public DefaultBatchHandlerImpl() {
+        super(BATCH_HANDLER_NAME);
+    }
+
+    @Override
+    public Batch newBatch() {
+        return initBatch();
+    }
+
+    protected Batch initBatch() {
+        return initBatch(null);
+    }
+
+    @Override
+    public Batch newBatch(String providedId) {
+        return initBatch(providedId);
+    }
+
+    @Override
+    public Batch getBatch(String batchId) {
+        TransientStore transientStore = getTransientStore();
+        Map<String, Serializable> batchEntryParams = transientStore.getParameters(batchId);
+
+        if (batchEntryParams == null) {
+            if (!hasBatch(batchId)) {
+                return null;
+            }
+            batchEntryParams = new HashMap<>();
+        }
+
+        String batchProvider = batchEntryParams.getOrDefault(PROVIDER_KEY, getName()).toString();
+        batchEntryParams.remove(PROVIDER_KEY);
+
+        if (getName().equals(batchProvider)) {
+            return new Batch(getName(), batchId, batchEntryParams, this);
+        }
+
+        return null;
+    }
+
+    protected boolean hasBatch(String batchId) {
+        return StringUtils.isNotEmpty(batchId) && getTransientStore().exists(batchId);
+    }
+
+    @Override
+    public void init(Map<String, String> configProperties) {
+        if (!containsRequired(configProperties)) {
+            throw new NuxeoException();
+        }
+
+        transientStoreName = configProperties.get(Config.TRANSIENT_STORE_NAME);
+        super.init(configProperties);
+    }
+
+    @Override
+    public boolean completeUpload(String batchId, String fileIndex, BatchFileInfo fileInfo) {
+        Batch batch = getBatch(batchId);
+        BatchFileEntry fileEntry = batch.getFileEntry(fileIndex, true);
+        return fileEntry.getFileSize() == fileInfo.getFileSize()
+                && Objects.equals(fileEntry.getMimeType(), fileInfo.getMimeType())
+                && Objects.equals(fileEntry.getBlob().getDigest(), fileInfo.getMd5());
```

Comment:
`getMimeType()`, `getDigest()` and `getMd5` return String, why not simply use `equals`?

Reply:
`Objects.equals` already handles the `null` cases